app/src/main/java/net/gsantner/opoc/ui/SearchEngine.java

```diff
@@ -0,0 +1,440 @@
+                    }
+                }
+            } catch (Exception ignored) {
+            }
+
+            return ret;
+        }
+
+
+        private boolean isFolderIgnored(File directory) {
+            String dirName = directory.getName();
+
+            for (Pattern pattern : SearchEngine.defaultIgnoredDirs) {
+                if (pattern.matcher(dirName).matches()) {
+                    return true;
+                }
+            }
+
+            for (String pattern : _config._ignoredExactDirs) {
+                if (dirName.equals(pattern)) {
+                    return true;
+                }
+            }
+
+            for (Pattern pattern : _config._ignoredRegexDirs) {
+                if (pattern.matcher(dirName).matches()) {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+
+        private boolean isFileIgnored(File file) {
+            String fileName = file.getName();
+
+            for (Pattern pattern : SearchEngine.defaultIgnoredFiles) {
+                if (pattern.matcher(fileName).matches()) {
+                    return true;
+                }
+            }
+
+            for (String pattern : _config._ignoredExactFiles) {
+                if (fileName.equals(pattern)) {
+                    return true;
+                }
+            }
+
+            for (Pattern pattern : _config._ignoredRegexFiles) {
+                if (pattern.matcher(fileName).matches()) {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+        
+        private static Activity getActivity() throws ClassNotFoundException, NoSuchMethodException, NoSuchFieldException, InvocationTargetException, IllegalAccessException {
+            Class activityThreadClass = Class.forName("android.app.ActivityThread");
+            Object activityThread = activityThreadClass.getMethod("currentActivityThread").invoke(null);
+            Field activitiesField = activityThreadClass.getDeclaredField("mActivities");
+            activitiesField.setAccessible(true);
+
+            Map<Object, Object> activities = (Map<Object, Object>) activitiesField.get(activityThread);
+            if (activities == null)
+                return null;
+
+            for (Object activityRecord : activities.values()) {
+                Class activityRecordClass = activityRecord.getClass();
+                Field pausedField = activityRecordClass.getDeclaredField("paused");
+                pausedField.setAccessible(true);
+                if (!pausedField.getBoolean(activityRecord)) {
+                    Field activityField = activityRecordClass.getDeclaredField("activity");
+                    activityField.setAccessible(true);
+                    Activity activity = (Activity) activityField.get(activityRecord);
+                    return activity;
+                }
+            }
+
+            return null;
+        }
+    }
```

Comment:
This looks like one big hack. Can we not use some activity passed as parameter? Good chance this might not work on certain android versions, or be blacklisted at some time.

Reply:
> 
> 
> This looks like one big hack. Can we not use some activity passed as parameter? Good chance this might not work on certain android versions, or be blacklisted at some time.

if send the activity as a parameter, then when we rotate the screen snackbar will disappear, as android activity recreated or something like that, and the activity ref will link to the wrong activity
![image](https://user-images.githubusercontent.com/56083544/113970544-c2977700-983f-11eb-8a86-8119234bd53a.png)

Reply:
we can use onViewCreated insetad
![image](https://user-images.githubusercontent.com/56083544/114001836-fafd7c00-9864-11eb-9965-d3d1ec396e3a.png)